### PR TITLE
rats-processors notebook app

### PR DESCRIPTION
**Pull request by elonp**
* Moved all existing sub-packages under rats.processors._legacy_subpackages.
* Refactored rats.pipelines_app as new subpackages under rats.processors.
* Significantly reduced the public interface of rats.processors.  We now have:
* * `rats.processors.types`: has a subset of what was is rats.processors.ux - the types that are needed for representing typed pipelines.
* * `rats.processors.NotebookApp`
* * `PipelineContainer`, `pipeline`, `task` - what's needed to define pipelines.
* * `Services` service ids exposed by `rats.processors`.
* * `example_pipelines` a sub package with example pipelines.

 `rats.processors.NotebookApp` can be created with no arguments.  It loads all containers that are associated with pyproject plugin hook `rats.processors_app_plugins`, and exposes a `run` method, a `display` method, and an `executable_pipelines` method exposing pipelines registered to `rats.processors.Services.Groups.EXECUTABLE_PIPELINES`.

### CODE REVIEW

File: rats-processors/src/python/rats/processors/_legacy_services_wrapper/_container.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    @apps.autoid_service
+    def pipeline_to_dot(self) -> IPipelineToDot:
+        return dag._viz.pipeline_to_dot
+
+
+class Services:
+    PIPELINE_RUNNER_FACTORY = apps.method_service_id(
+        LegacyServicesWrapperContainer.pipeline_runner_factory
+    )
+    PIPELINE_TO_DOT = apps.method_service_id(LegacyServicesWrapperContainer.pipeline_to_dot)
```
[the remark this answers is not in the document]
Comment (elonp) [reply rats-processors/src/python/rats/processors/_legacy_services_wrapper/_container.py]: These services are part of the public interface of _legacy_services_wrapper, but the services and the subpackage (_legacy_services_wrapper) are not exposed as part of the public interface of rats.processors.

The _notebook_app subpackage uses these services, so they need to be exposed by _legacy_services_wrapper. 

Later on I'll need to use them in cli, and we might also decide to expose them publicly.  If we do, I think we'll have rats.processors._services import them from _legacy_services_wrapper and copy the service ids to rats.processors._services.Services, which is exposed as rats.processors.Services.

File: rats-processors/src/python/rats/processors/__init__.py [hunk 2]
```diff
@@ -1,11 +1,19 @@
...
-    "display_dag",
-    "TrainAndEvalBuilders",
-    "ScatterGatherBuilders",
-    "Pipeline",
+    "IPipelineRegistry",
+    "NotebookApp",
+    "PipelineContainer",
+    "pipeline",
+    "types",
+    "Services",
```
[the remark this answers is not in the document]
Comment (elonp) [reply rats-processors/src/python/rats/processors/__init__.py:17]: We're exposing `Services`.  Currently it only has one id in it: `Services.Groups.EXECUTABLE_PIPELINES_REGISTRY`.  I believe placing `Groups` under `Services` is the pattern @ms-lolo suggested.
I expected we'll add more service ids to `Services` and `Services.Groups` soon.

File: rats-processors/src/python/rats/processors/_legacy_services_wrapper/_container.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (jzazo) [reply rats-processors/src/python/rats/processors/_legacy_services_wrapper/_container.py]: ok, sounds good.

File: rats-processors/src/python/rats/processors/_pipeline_registry/_container.py [hunk 3]
```diff
@@ -0,0 +1,19 @@
+from itertools import chain
+
+from rats import apps
+
+from ._pipeline_registry import PipelineRegistry
+from ._services import Groups, Services
+
+
+class Container(apps.AnnotatedContainer):
```
[the remark this answers is not in the document]
Comment (elonp) [reply rats-processors/src/python/rats/processors/_pipeline_registry/_container.py:9]: What is a `ProcessorsContainer`?  I don't think we have that defined anywhere.
This is a `Container` of services provided by the subpackage `_pipeline_registry`.  

Following our naming discussion from a few days ago, I am moving away from the custom of including the package name in the container class name.

**Pull request merged**